document/src/main/java/com/ritense/document/service/impl/JsonSchemaDocumentSearchService.java

```diff
@@ -190,25 +190,15 @@ private Page<JsonSchemaDocument> search(QueryWhereBuilder queryWhereBuilder, Pag
         return new PageImpl<>(documents, pageable, count(queryWhereBuilder));
     }
 
-    private Long count(
-        QueryWhereBuilder queryWhereBuilder
-    ) {
+    private Long count(QueryWhereBuilder queryWhereBuilder) {
         final CriteriaBuilder cb = entityManager.getCriteriaBuilder();
         final CriteriaQuery<Long> countQuery = cb.createQuery(Long.class);
         Root<JsonSchemaDocument> countRoot = countQuery.from(JsonSchemaDocument.class);
-        countQuery.select(cb.count(countRoot));
+        countQuery.select(cb.countDistinct(countRoot));
         queryWhereBuilder.apply(cb, countQuery, countRoot);
 
         // TODO: Should be turned into a subquery, and then do a count over the results from the subquery.
-        List<Long> countResultList = entityManager.createQuery(countQuery).getResultList();
-
-        long count = 0L;
-
-        if (!countResultList.isEmpty()) {
-            count = countResultList.size();
-        }
-
-        return count;
+        return entityManager.createQuery(countQuery).getSingleResult();
```

Comment:
This changes the outcome. But there isn't a single test that breaks. 
I wonder why a count query used `.size();` 

@ivo-ritense do you know?

Reply:
It's because we no longer do a group by for count queries. This means we get a single result instead of a list of results, so we no longer need the `size()`. The `countDistinct` is used to prevent duplicate results caused by the join on other tables. This is closer to the 'proper' SQL query that should be used for something like this.